src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessNextDataSetVersionFunctionTests.cs

```diff
@@ -225,72 +228,80 @@ public async Task Success_Mappings()
                         .ToList()))
                 .GenerateList();
-
+            
             await AddTestData<PublicDataDbContext>(context =>
                 context.LocationMetas.AddRange(initialLocationMeta));
 
             await CreateMappings(instanceId);
 
-            var mappings = GetDbContext<PublicDataDbContext>()
-                .DataSetVersionMappings
-                .Single();
+            var mappings = GetDataSetVersionMapping(nextVersion);
 
             Assert.Equal(initialVersion.Id, mappings.SourceDataSetVersionId);
             Assert.Equal(nextVersion.Id, mappings.TargetDataSetVersionId);
             Assert.False(mappings.LocationMappingsComplete);
 
             var expectedLocationMappingsFromSource = initialLocationMeta
-                .OrderBy(levelMeta => levelMeta.Level)
-                .Select(levelMeta => new LocationLevelMappings
-                {
-                    Level = levelMeta.Level,
-                    Mappings = levelMeta
-                        .Options
-                        .OrderBy(option => option.Label)
-                        .Select(option => new LocationOptionMapping
-                        {
-                            CandidateKey = null,
-                            Type = MappingType.None,
-                            Source = new LocationOption
-                            {
-                                Key = $"{option.Label} :: {option.ToRow().GetRowKey()}",
-                                Label = option.Label
-                            }
-                        })
-                        .ToList()
-                })
-                .ToList();
+                .ToDictionary(
+                    levelMeta => levelMeta.Level,
+                    levelMeta => new LocationLevelMappings
+                    {
+                        Mappings = levelMeta
+                            .Options
+                            .ToDictionary(
+                                keySelector: option => $"{option.Label} :: {option.ToRow().GetRowKey()}",
+                                elementSelector: option => new LocationOptionMapping
+                                {
+                                    CandidateKey = null,
+                                    Type = MappingType.None,
+                                    Source = new LocationOption(option.Label)
+                                    {
+                                        Code = option.ToRow().Code,
+                                        OldCode = option.ToRow().OldCode,
+                                        Urn = option.ToRow().Urn,
+                                        LaEstab = option.ToRow().LaEstab,
+                                        Ukprn = option.ToRow().Ukprn
+                                    }
+                                })
+                    });
 
             // There should be 5 levels of mappings when combining all the source and target levels. 
```

Comment:
think this will be 6 now won't it?

Reply:
It's still 5 because of the shared levels between the 2 data set versions.

Reply:
Ah yes! sorry, I didn't realise `School` was in the new data set